assets/app/js/currencyConverter.js

```diff
@@ -8,56 +8,68 @@ $(document).ready(function () {
         headers: {
             'X-Mashape-Key': 'Bs5BvTwMeNmshIVgyxatfWRfPMkNp1Dmi30jsnLUNZ8zyDyBW8'
         }
-    })
-	.done(function (data) {
-
-	    for (var i = 0; i < data.length; i++) {
-	        var elem = document.createElement('option');
-	        $('.currency-selecter')
-			.append($(elem)
-			.attr('value', data[i].id).text(data[i].description.slice(4)));
+    }).done(function (data) {
+
+        var i = 0,
+            len = data.length,
+            option,
+            option2,
+            datum,
+            docFrag = document.createDocumentFragment(),
+            docFrag2 = document.createDocumentFragment(),
+            selectEl = $('.currency-selecter');
+
+	    for (i = 0; i < len; i++) {
+            datum = data[i];
+            option = document.createElement('option');
+            option.value = datum.id;
+            option.text = datum.description.slice(4);
+            option2 = document.createElement('option');
+            option2.value = option.value;
+            option2.text = option.text;
+            docFrag.appendChild(option);
+            docFrag2.appendChild(option2);
 	    }
+
+        selectEl[0].appendChild(docFrag);
+        selectEl[1].appendChild(docFrag2);
 	});
 
     $('.currency-From').on('keyup', function (e) {
```

Comment:
you can avoid a lot of this (setting timeout, clearing timeout by using debounce